### ISSUE TITLE
Bug: APIGatewayRestResolver - openAPI metadata responses / tags on @router fails with unhashable type: dict 

### ISSUE THREAD

**Issue by MCR2019** [awslabs/aws-lambda-powertools-python#3506]
### Expected Behaviour

When adding `responses` or `tags` metadata to a router this should behave in the same was as when it is added to the app.

### Current Behaviour

When the routers are added to the app the metadata is used as a key. As the metadata fields for `responses` and `tags` are Dict and List then this produces the following error:

```  
File "/Users/my_path/venv/lib/python3.10/site-packages/aws_lambda_powertools/event_handler/api_gateway.py", line 2160, in register_route
    self._routes_with_middleware[route_key] = []
TypeError: unhashable type: 'dict'
```

### Code snippet

```python
from aws_lambda_powertools.event_handler import APIGatewayRestResolver
from aws_lambda_powertools.event_handler.api_gateway import Router

router = Router()

app = APIGatewayRestResolver(enable_validation=True)
app.include_router(router)


@router.put(
    "/example-resource",
    responses={200: {"description": "Todo item found"}},
    tags=["Examples"]
)
def put():
    """"""


if __name__ == "__main__":
    """"""
```

### Steps to Reproduce

- Create APIGatewatRestResolver.
- Create api_gateway Router.
- Add method to Router with `responses` or `tags` metadata.
- Add Router to app.

### Powertools for AWS Lambda (Python) version

2.29.1

### AWS Lambda function runtime

3.10

### Packaging format used

PyPi

**Comment by leandrodamascena** [comment]
Hello @MCR2019! Thanks for reporting this bug! I was able to reproduce the error and we are working to fix it.
As it appears critical for those who are using the `Router` object and now integrating with the new OpenAPI utility, we will fix it as soon as possible and plan to release a patch by tomorrow.

We apologize for this bug. OpenAPI + Swagger are new utilities and we try our best to create tests and keep our code coverage high. Unfortunately, some things go through and we can't simulate them until someone reports it :disappointed: 

Thank you.

**Comment by rubenfonseca** [comment]
Working at this now!

**Comment by rubenfonseca** [comment]
@MCR2019 thank you again for reporting  this, we've just released 2.30.1 that should fix the problem you found!

**Comment by MCR2019** [comment]
@rubenfonseca - Thank you for fixing so quickly!
